x-pack/plugins/synthetics/server/legacy_uptime/lib/saved_objects/migrations/monitors/8.8.0.ts

```diff
@@ -118,3 +126,102 @@ const omitZipUrlFields = (fields: BrowserFields) => {
 
   return formatSecrets(validationResult.decodedMonitor);
 };
+
+const updateThrottlingFields = (
+  doc: SavedObjectUnsanitizedDoc<
+    SyntheticsMonitorWithSecrets &
+      Partial<{
+        [LegacyConfigKey.THROTTLING_CONFIG]: string;
+        [LegacyConfigKey.IS_THROTTLING_ENABLED]: boolean;
+        [LegacyConfigKey.DOWNLOAD_SPEED]: string;
+        [LegacyConfigKey.UPLOAD_SPEED]: string;
+        [LegacyConfigKey.LATENCY]: string;
+      }>
+  >,
+  logger: SavedObjectMigrationContext
+) => {
+  try {
+    const { attributes } = doc;
+
+    const migrated = {
+      ...doc,
+      attributes: {
+        ...doc.attributes,
+        [ConfigKey.CONFIG_HASH]: '',
+      },
+    };
+
+    const isThrottlingEnabled = attributes[LegacyConfigKey.IS_THROTTLING_ENABLED];
+    if (isThrottlingEnabled) {
+      const download = String(attributes[LegacyConfigKey.DOWNLOAD_SPEED])!;
+      const upload = String(attributes[LegacyConfigKey.UPLOAD_SPEED])!;
+      const latency = String(attributes[LegacyConfigKey.LATENCY])!;
+
+      // @ts-ignore
+      migrated.attributes[ConfigKey.THROTTLING_CONFIG] = getMatchingThrottlingConfig(
+        download,
+        upload,
+        latency
+      );
+    } else {
+      // @ts-ignore
+      migrated.attributes[ConfigKey.THROTTLING_CONFIG] =
+        PROFILES_MAP[PROFILE_VALUES_ENUM.NO_THROTTLING];
+    }
+
+    // filter out legacy throttling fields
+    return {
+      ...migrated,
+      attributes: omit(migrated.attributes, [
+        LegacyConfigKey.THROTTLING_CONFIG,
+        LegacyConfigKey.IS_THROTTLING_ENABLED,
+        LegacyConfigKey.DOWNLOAD_SPEED,
+        LegacyConfigKey.UPLOAD_SPEED,
+        LegacyConfigKey.LATENCY,
+      ]),
+    };
+  } catch (e) {
+    logger.log.warn(
+      `Failed to migrate throttling fields from legacy Synthetics monitor: ${e.message}`
+    );
+    const { attributes } = doc;
+
+    // @ts-ignore
+    attributes[ConfigKey.THROTTLING_CONFIG] = PROFILES_MAP[PROFILE_VALUES_ENUM.DEFAULT];
+
+    return {
+      ...doc,
+      attributes: omit(attributes, [
+        LegacyConfigKey.THROTTLING_CONFIG,
+        LegacyConfigKey.IS_THROTTLING_ENABLED,
+        LegacyConfigKey.DOWNLOAD_SPEED,
+        LegacyConfigKey.UPLOAD_SPEED,
+        LegacyConfigKey.LATENCY,
+      ]),
+    };
+  }
+};
+
+const getMatchingThrottlingConfig = (download?: string, upload?: string, latency?: string) => {
```

Comment:
These shouldn't have to be optional, by the time we get to this file we should have confidence that all fields are filled out, download, upload, and latency. See my comment on line 156.